openwisp_monitoring/device/api/serializers.py

```diff
@@ -20,6 +26,79 @@ class Meta:
         model = DeviceMonitoring
 
 
+class WifiClientSerializer(FilterSerializerByOrgManaged, ValidatedModelSerializer):
```

Comment:
`FilterSerializerByOrgManaged` mixin filters the choices in the queryset of the fields with respect the value of `organization_field`. We don't need `FilterSerializerByOrgManaged` here because there is no field which is tied to an organization.

Reply:
Yes, we should do that. We should only check if the user has permissions to manage `WifiClient`, if so they can perform required operation on the object. 

`DjangoModelPermission` should get us covered in this scenario.

Reply:
Thanks !

Reply:
@pandafy Some more questions:
1. It is still relevant to add a test for `test_wifi_clients_multi_tenancy`? (because now anyone can view this endpoint with right permission)?
2. Do you have any idea about [above error](https://github.com/openwisp/openwisp-monitoring/pull/382#issuecomment-1156348391) **(MultipleObjectsReturned)**  in `wifi-session` endpoint?

Reply:
One WiFIClient can have multiple WiFiSession. The WifiSession can be with devices of different organizations. I don't think the `organization_filed` implementation will work here.

Reply:
@nemesisdesign I've fixed this in [56e87f1](https://github.com/openwisp/openwisp-monitoring/pull/382/commits/56e87f1b963c445e3e63bf501449bb569841ac23)  & also tested manually, I think now it is working as you've mentioned above. Now trying to fix https://github.com/openwisp/openwisp-monitoring/pull/382#issuecomment-1156348391 (MultipleObjectsReturned) in `wifi-session` endpoint.

Reply:
@Aryamanz29 can you write a test for the issue you are facing (`MultipleObjectsReturned`). It will help me easily debug that and help you.

Reply:
@pandafy Please see https://github.com/openwisp/openwisp-monitoring/pull/382/commits/aa9a0f5c18c341700b379610f9e1b62b14910063 I have added both tests with **non-superuser administrator group** and with **superuser**.

Reply:
> @nemesisdesign I've fixed this in [56e87f1](https://github.com/openwisp/openwisp-monitoring/pull/382/commits/56e87f1b963c445e3e63bf501449bb569841ac23) & also tested manually, I think now it is working as you've mentioned above. Now trying to fix [#382 (comment)](https://github.com/openwisp/openwisp-monitoring/pull/382#issuecomment-1156348391) (MultipleObjectsReturned) in `wifi-session` endpoint.

While working on this [67d0235](https://github.com/openwisp/openwisp-monitoring/pull/382/commits/67d02356ad6da80195199f72d62f99e5dd4f2d92) I found that we can only filter **wificlient** **APIVIEWS** according to user's managed organization only when clients have session (since filtering is based on  **wifisession__device__organization**). Let say if any user **(org A)** created a new **wificlient** using API endpoint and that client doesn't have any session, then that client will be visible to **org B** user.